source/ImageSet.cpp

```diff
@@ -222,6 +238,20 @@ void ImageSet::ValidateFrames() noexcept(false)
 				+ " extra frames for the @2x sprite will be ignored.");
 		paths[1].resize(paths[0].size());
 	}
+	// Drop any mask paths that will not be used.
+	if(paths[2].size() > paths[0].size())
+	{
+		Logger::LogError(prefix + to_string(paths[2].size() - paths[0].size())
+				+ " extra frames for the mask sprite will be ignored.");
+		paths[2].resize(paths[0].size());
+	}
+	// Drop any @2x mask paths that will not be used.
+	if(paths[3].size() > paths[0].size())
+	{
+		Logger::LogError(prefix + to_string(paths[3].size() - paths[0].size())
+				+ " extra frames for the @2x mask sprite will be ignored.");
+		paths[3].resize(paths[0].size());
+	}
```

Comment:
The duplication here is not great, can you avoid it?

Reply:
did it